src/backend/FileStorageSystem.js

```diff
@@ -13,85 +17,156 @@ const doGetLocalStorageSourceIdNames = () => convertStrValueOrDefault(
+  const backendModeSignedInStatus = store.getState().backendModeSignedInStatus;
+  if (backendModeSignedInStatus === BACKEND_MODE_SIGNED_IN_STATUS.LOCAL_STORAGE) {
+    return Object.fromEntries(Object.entries(doGetLocalStorageSourceIdNames()).map(
+      ([sourceId, sourceName]) => [
+        sourceId,
+        { name: sourceName, viewsList: doGetLocalStorageSourceViewsList(sourceId) },
+      ]
+    ));
+  } else if (backendModeSignedInStatus === BACKEND_MODE_SIGNED_IN_STATUS.USER_SIGNED_IN) {
+    const token = Cookies.get(ACCESS_TOKEN_COOKIE_KEY);
+    try {
+      const response = await fetchWithTimeout(
+        SERVER_BASE_URL + `documents`,
+        { headers: new Headers({ 'Set-Cookie': `token=${token}` }) },
+      );
+      if (response.ok) {
+        const responseValue = await response.json();
+        return Object.fromEntries(Object.entries(responseValue).map( // TODO just return directly if views map not list
+          ([sourceId, { name, views }]) => [
+            sourceId,
+            {
+              name,
+              viewsList: Object.fromEntries(views.map(
+                ({ id: viewId, name: viewName, type }) => [viewId, { name: viewName, type }]
+              )),
+            },
+          ]
+        ));
+      }
+    } catch (e) { console.log(e); }
+  }
+  return null;
+};
 
 // API-ish
 export const doFileNamesSearch = (filesList, search) => Object.fromEntries(
-  Object.entries(filesList).map(([sourceId, { name: sourceName, viewIdNames }]) => [
+  Object.entries(filesList).map(([sourceId, { name: sourceName, viewsList }]) => [
     sourceId,
     {
       name: sourceName,
-      viewIdNames: search
-        ? Object.fromEntries(Object.entries(viewIdNames).filter(
-            ([_viewId, { viewName }]) => viewName.includes(search)
+      viewsList: search
+        ? Object.fromEntries(Object.entries(viewsList).filter(
+            ([_viewId, { name: viewName }]) => viewName.includes(search)
           ))
-        : viewIdNames,
+        : viewsList,
     },
   ]).filter(
-    ([_sourceId, { name: sourceName, viewIdNames }]) => !search ||
-      Object.keys(viewIdNames).length > 0 || sourceName.includes(search)
+    ([_sourceId, { name: sourceName, viewsList }]) => !search ||
+      Object.keys(viewsList).length > 0 || sourceName.includes(search)
   )
 );
 
 export const countNumFiles = filesList => Object.entries(filesList)
-  .reduce((count, [_sourceId, { viewIdNames }]) => count + 1 + viewIdNames.length, 0);
+  .reduce((count, [_sourceId, { viewsList }]) => count + 1 + Object.keys(viewsList).length, 0);
 
 const VIEW_CONTENT_SPEC_LOCAL_STORAGE_KEY_PREFIX = 'viewContentSpec_';
 
 const SOURCE_CONTENT_LOCAL_STORAGE_KEY_PREFIX = 'sourceContent_';
 
+const EMPTY_SOURCE_CONTENT = { type: 'doc', content: [{ type: 'paragraph', attrs: { tags: {} } }] };
+
 // API
-export const doSaveSourceContent = (sourceId, value) => {
+export async function doSaveSourceContent(value, sourceId, createSourceName) {
```

Comment:
I'll add a more general todo for what we should do with the localstorage. Right now it's being treated as a completely separate mode rather than as a cache. If the user is in non local storage mode, it won't save if there connection to the backend fails.